test/integration-test.py

```diff
@@ -621,34 +619,51 @@ def main():
                         help="run load-generator")
     parser.add_argument('--filter', dest="test_case_filter", action="store",
                         help="Regex filter for test cases")
+    parser.add_argument('--skip-setup', dest="skip_setup", action="store_true",
+                        help="skip integration test setup")
     # allow any ACME client to run custom command for integration
     # testing (without having to implement its own busy-wait loop)
     parser.add_argument('--custom', metavar="CMD", help="run custom command")
     parser.set_defaults(run_all=False, run_certbot=False, run_chisel=False,
-        run_loadtest=False, test_case_filter="")
+        run_loadtest=False, test_case_filter="", skip_setup=False)
     args = parser.parse_args()
 
     if not (args.run_all or args.run_certbot or args.run_chisel or args.run_loadtest or args.custom is not None):
         raise Exception("must run at least one of the letsencrypt or chisel tests with --all, --certbot, --chisel, --load or --custom")
 
-    now = datetime.datetime.utcnow()
-    seventy_days_ago = now+datetime.timedelta(days=-70)
-    if not startservers.start(race_detection=True, fakeclock=fakeclock(seventy_days_ago)):
-        raise Exception("startservers failed (mocking seventy days ago)")
-    setup_seventy_days_ago()
-    startservers.stop()
-
-    now = datetime.datetime.utcnow()
-    twenty_days_ago = now+datetime.timedelta(days=-20)
-    if not startservers.start(race_detection=True, fakeclock=fakeclock(twenty_days_ago)):
-        raise Exception("startservers failed (mocking twenty days ago)")
-    setup_twenty_days_ago()
-    startservers.stop()
+    if not args.skip_setup:
+
+        now = datetime.datetime.utcnow()
+        seventy_days_ago = now+datetime.timedelta(days=-70)
+        if not startservers.start(race_detection=True, fakeclock=fakeclock(seventy_days_ago)):
+            raise Exception("startservers failed (mocking seventy days ago)")
+        global caa_client
+        caa_client = chisel.make_client()
+        setup_seventy_days_ago()
+        startservers.stop()
+
+        now = datetime.datetime.utcnow()
+        twenty_days_ago = now+datetime.timedelta(days=-20)
+        if not startservers.start(race_detection=True, fakeclock=fakeclock(twenty_days_ago)):
+            raise Exception("startservers failed (mocking twenty days ago)")
+        setup_twenty_days_ago()
+        startservers.stop()
+
+    
+    if caa_client is None:
+        # start boulder without a account_uri so we can create an account then tear
```

Comment:
It looks like this stanza is here so that even when you run with `--skip-setup`, the test case for `accounturi.good-caa-reserved.com.` (part of `test_caa`) still passes. However, aren't there also a bunch of other tests that fail without setup? It seems weird to special-case `test_caa`. Instead I think it's reasonable to just say that certain tests are expected to fail if you skip setup, caveat emptor, and use the filter flag. :-)

Reply:
Works for me.